### PR TITLE
chore(argo-workflows): Update docs to mention why CRDs are in templates folder

### LINKED ISSUE #1785: Move CRDs back to root folder
### Is your feature request related to a problem?

With `crds` in the templates folder, we can't install `argo-workflows` and resources that use its custom resources (eg. `WorkflowTemplate`) on a single helm install.
From what I understood, this was done to avoid problems with updates to the CRDs when the chart was already installed. 

### Related helm chart

argo-workflows

### Describe the solution you'd like

According to [Helm best practices](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/), the custom resources should be in a folder called `crds` on the root directory of the chart.
That would allow the installation and usage of the CRDs on a single run.
To solve the problem with CRD updates, I'd suggest we create a new version of the CRD with the changes when needed.

### Describe alternatives you've considered

_No response_

### Additional context

When installing argo-workflows and a resource of kind WorkflowTemplate in a single run:
```bash
Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "WorkflowTemplate" in version "argoproj.io/v1alpha1"
helm.go:84: [debug] unable to recognize "": no matches for kind "WorkflowTemplate" in version "argoproj.io/v1alpha1"
```

### PR DESCRIPTION
**Pull request by jmeridth**
Closes #1785

Borrowed `Custom resource definition` from [argo-cd helm chart README](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/README.md#custom-resource-definitions) and added it to argo-workflows helm chart README

Added a more generalized section to the base `README.md` about CRDs.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Comment by jmeridth** [comment]
Since this is in the four primary helm charts (argo-cd, argo-workflows, argo-events, argo-rollouts), should we just have this section in our base README.md?

**Comment by yu-croco** [comment]
> Since this is in the four primary helm charts (argo-cd, argo-workflows, argo-events, argo-rollouts), should we just have this section in our base README.md?

This sounds nice!  :)

File: charts/argo-workflows/README.md [hunk 1]
```diff
@@ -13,6 +13,21 @@ A few options are:
 - Manually create a ServiceAccount in the Namespace in which your release will be deployed w/ appropriate bindings to perform this action and set the `serviceAccountName` field in the Workflow spec
 - Augment the `default` ServiceAccount permissions in the Namespace in which your Release is deployed to have the appropriate permissions
 
+### Custom resource definitions
+
+Some users would prefer to install the CRDs _outside_ of the chart. You can disable the CRD installation of this chart by using `--set crds.install=false` when installing the chart.
+
+Helm cannot upgrade custom resource definitions in the `<chart>/crds` folder [by design](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations). Starting with 3.4.0 (chart version 0.19.0), the CRDs have been moved to `<chart>/templates` to address this design decision.
+
+If you are using Argo CD chart version prior to 3.4.0 (chart version 0.19.0) or have elected to manage the Argo CD CRDs outside of the chart, please use `kubectl` to upgrade CRDs manually from [templates/crds](templates/crds/) folder or via the manifests from the upstream project repo:
```
Comment (vladlosev) [inline charts/argo-workflows/README.md:22]:
```suggestion
If you are using Argo Workflows chart version prior to 3.4.0 (chart version 0.19.0) or have elected to manage the Argo Workflows CRDs outside of the chart, please use `kubectl` to upgrade CRDs manually from [templates/crds](templates/crds/) folder or via the manifests from the upstream project repo:
```

File: charts/argo-workflows/README.md.gotmpl [hunk 2]
```diff
@@ -13,6 +13,21 @@ A few options are:
 - Manually create a ServiceAccount in the Namespace in which your release will be deployed w/ appropriate bindings to perform this action and set the `serviceAccountName` field in the Workflow spec
 - Augment the `default` ServiceAccount permissions in the Namespace in which your Release is deployed to have the appropriate permissions
 
+### Custom resource definitions
+
+Some users would prefer to install the CRDs _outside_ of the chart. You can disable the CRD installation of this chart by using `--set crds.install=false` when installing the chart.
+
+Helm cannot upgrade custom resource definitions in the `<chart>/crds` folder [by design](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations). Starting with 3.4.0 (chart version 0.19.0), the CRDs have been moved to `<chart>/templates` to address this design decision.
+
+If you are using Argo CD chart version prior to 3.4.0 (chart version 0.19.0) or have elected to manage the Argo CD CRDs outside of the chart, please use `kubectl` to upgrade CRDs manually from [templates/crds](templates/crds/) folder or via the manifests from the upstream project repo:
```
Comment (vladlosev) [inline charts/argo-workflows/README.md.gotmpl:22]:
```suggestion
If you are using Argo Workflows chart version prior to 3.4.0 (chart version 0.19.0) or have elected to manage the Argo Workflows CRDs outside of the chart, please use `kubectl` to upgrade CRDs manually from [templates/crds](templates/crds/) folder or via the manifests from the upstream project repo:
```

File: charts/argo-workflows/README.md (hunk 1 above)
Comment (jmeridth) [reply charts/argo-workflows/README.md]: Saw this, sorry, I force pushed the fix.  Would have merged your offering but missed it. Thank you though

**Comment by jmeridth** [comment]
@yu-croco base `README.md` updated also.

**Comment by mkilchhofer** [comment]
👍

**Review by mkilchhofer — APPROVED** [review]
👍

**Review by vladlosev — APPROVED** [review]
LGTM

**Pull request merged**